src/eckit/geo/util/reduced_classical_pl.cc

```diff
@@ -0,0 +1,1366 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#include <map>
+#include <string>
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/geo/Cache.h"
+#include "eckit/geo/util.h"
+
+
+namespace eckit::geo::util {
+
+
+static const std::map<size_t, pl_type> CLASSICAL_PLS{
```

Comment:
I think this is analogous to atlas. There this is done to avoid every MPI rank to load the same data file upon loading a grid. This issue was also leading to eccodes' possibility to compile its definitions into a shared library.
For eckit::geo, I guess this could also become a configurable option.

Reply:
Ah ah yes, this comes straight from Atlas; the main difference is the value_type is long (pl_type fordwards this type), which matches the GRIB format, hence no conversion takes place (so there should be a benefit here). If we put it in a source file it will pay the penalty in every instantiation, so no. Also, this is very small in footprint (I haven't calculated it but my estimate is... peanuts :-)